guardian/support-service-lambdas
# Zuora <-> salesforce link remover - create state machine

## What does this change?
**Parent Delivery Trello ticket** [Convert Billing Account Remover Lambda from scala to typescript](https://trello.com/c/ne8VPaj0/61-convert-billing-account-remover-lambda-from-scala-to-typescript-l)

Introduce a new State Machine to orchestrate the execution of lambdas created in PRs:

- get billing accounts from salesforce [#2347](https://github.com/guardian/support-service-lambdas/pull/2347)
- update billing account in zuora [#2352](https://github.com/guardian/support-service-lambdas/pull/2352)
- update sf billing accounts [#2357](https://github.com/guardian/support-service-lambdas/pull/2357)

<img width="568" alt="Screenshot 2024-07-10 at 11 59 27" src="https://github.com/guardian/support-service-lambdas/assets/36296660/ba233538-ea85-43f7-a7f6-ff6f63bc67c7">


### State Machine, with example inputs and outputs
<img width="1038" alt="Screenshot 2024-07-10 at 11 33 03" src="https://github.com/guardian/support-service-lambdas/assets/36296660/0d283354-56ed-4bbd-962b-029b577a9cdc">


## How to test
- Create test data in Zuora that will be retrieved by the start query in lambda that queries from Salesforce. Update the test query as needed.
- execute the state machine
- Ensure the crmIds have been removed from the billing accounts in Zuora, and the write backs to Salesforce have been successful (i.e. GDPR_Removal_Attempts__c on billing account records has been incremented by 1). N.B. z360 should then delete the billing accounts in Salesforce a few seconds later, so this can be verified by checking deleted records.
